### PR TITLE
docs(google-cloud-storage): Document support for dual region buckets

### PR DESCRIPTION
**Pull request by SandeepTuniki**
Addresses feature request raised in #17905 

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #17905

### CODE REVIEW

File: google-cloud-storage/samples/storage_create_bucket_dual_region.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START storage_create_bucket_dual_region]
+require "google/cloud/storage"
+
+def storage_create_bucket_dual_region bucket_name:, region_1:, region_2:
```
Comment (dazuma) [inline google-cloud-storage/samples/storage_create_bucket_dual_region.rb:18]: Can you start using classes to namespace the sample methods? (I know the other storage samples don't do this yet, but we have to start sometime, and now is better than later.)

**Review by dazuma — CHANGES_REQUESTED** [review]
This doesn't look like a `feat:`. I don't see any new functionality in the client library. In fact, the only user-visible change I can see here is a documentation change. Can you adjust the commit message accordingly? Also, remember, the commit message should reflect what should appear in the user-visible changelog for the google-cloud-storage gem, so it should talk about library functionality and documentation, but not samples because they are not part of the gem.

File: google-cloud-storage/samples/storage_create_bucket_dual_region.rb (hunk 1 above)
Comment (SandeepTuniki) [reply google-cloud-storage/samples/storage_create_bucket_dual_region.rb]: Namespaced the sample with a class.

**Comment by SandeepTuniki** [comment]
Thanks @dazuma for the review. I made the requested changes.

**Review by dazuma — APPROVED** [review]
LGTM

**Review by SandeepTuniki — COMMENTED** [review]
docs(google-cloud-storage): Document support for dual region buckets

**Pull request merged**